presentation/src/main/java/bg/government/virusafe/mvvm/fragment/AbstractFragment.kt

```diff
@@ -270,6 +274,23 @@ abstract class AbstractFragment<B : ViewDataBinding, VM : AbstractViewModel> :
 		}
 	}
 
+	fun showAgreementsDialog(
+		title: String,
+		description: String,
+		agreement: Agreement,
+		showAgreeBtn: Boolean = false,
+		onClick: OnDialogButtonListener? = null
+	) {
+		val dialog = AgreementsDialog.newInstance(title, description, agreement, showAgreeBtn)
+		val fm: FragmentManager? = activity?.supportFragmentManager
+		onClick?.let {
```

Comment:
Also since in the dialog class the click listener is set to null, tou can change the param in the setClickListener() function to be nullable and remove this check.